docs/developers/ROADMAP_0_3.md

```diff
@@ -0,0 +1,76 @@
+
+- **Eliminate all known bugs**. See the [`bug` label](https://github.com/napari/napari/labels/bug) for a current list of known bugs. We also want to make it easier for users to report bugs ([#1090](https://github.com/napari/napari/issues/1090)). Additionally, when bugs are encountered, we want to examine whether an improved software architecture could have prevented them. For example, we are undertaking a refactor to centralize event handling within napari and avoid circular, out of order, or repeated function calls ([#1040](https://github.com/napari/napari/issues/1040)).
+
+- **Improve support for annotation** of points [#858](https://github.com/napari/napari/issues/858), shapes [#177](https://github.com/napari/napari/issues/177), labels, and images [#209](https://github.com/napari/napari/issues/209) including text rendering [#600](https://github.com/napari/napari/pull/600).
+
+- Add **linked multi-canvas support** ([#760](https://github.com/napari/napari/issues/760)) to allow orthogonal views, or linked views of two datasets, for example to select keypoints for image alignment, or simultaneous 2D slices with 3D rendering.
+
+
+### A downloadable application with basic plugin management
+
+- Distribute **a bundled application for each major OS** [#496](https://github.com/napari/napari/pull/496) to allow scientists to use napari without requiring a Python development environment, which can be difficult to install and maintain. The bundle should contain the necessary machinery to add plugins [#1074](https://github.com/napari/napari/issues/1074).
+
+- Support for reader plugins [#937](https://github.com/napari/napari/pull/937) and writer plugins [#1068](https://github.com/napari/napari/issues/1068) to allow **viewing of domain-specific data and saving of annotations**. For more details see the [`plugins` label](https://github.com/napari/napari/labels/plugins) on our repository.
+
+### Accessible documentation, tutorials, and demos
+
+- Improve our website [napari.org](https://napari.org) to provide easy access to all napari related materials, including the [**four types of documentation**](https://www.divio.com/blog/documentation/): learning-oriented tutorials, goal-oriented how-to-guides or galleries, understanding-oriented explanations (including developer resources), and a comprehensive API reference. See [#764](https://github.com/napari/napari/issues/764) and the [`documentation` label](https://github.com/napari/napari/labels/documentation) on the napari repository for more details.
+
+- Add a **napari human interface guide** for plugin developers, akin to [Apple's Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/). We want such a guide to promote best practices and ensure that plugins provide a consistent user experience.
+
+
+### Other improvements
```

Comment:
I would eliminate this section as it just seems like leftover items and doesn't fit into the three goals above.
You could change the first goal to "Make the data viewing and annotation capabilities bug-free, fast _**and easy to use**_**." and then we can add the "Improve the **user interface and design** of the viewer." in there.

"Improve our unit testing" could also go into the first goal because of "bug-free".

"layer groups" is the odd one out, but you could argue that it could then fall under "easy to use".

Reply:
I agree, I like those suggestions. @jni?

Reply:
So, Nick and I debated about this for some time. To be honest, I would favour dropping CI, serialization, and headless. The other two, focused on user friendliness, I'm happy to move along with a relabel of the top-level goal. (Though together they might merit their own subsection.)

My perspective is that the document is supposed to outline priorities and focus effort. Its purpose becomes a bit more muddled if *everything* is a priority — which it's not too far off from already!

But, Nick wanted to include these items so we don't lose track of them. Fine. But then we should have some sort of indication that they are less critical/urgent/should come after the others. Hence the "other improvements".

Additionally, another idea I had for the document is that it should focus on the *user experience*. So, words like *refactor* and *software architecture* should be neatly buried, not front and centre — they should only serve the other, user-facing goals. I think "improve CI" falls in that category — most users don't even know what CI is.

Now, in terms of the items discussed:

- CI: not user facing at all. This is a problem we face as developers and we should expend effort getting it right, but I don't think it needs to be on the roadmap. As developers, we directly feel the pain, and we are (sometimes) motivated to go fix it in the course of pursuing the other goals.
- layer groups: I've often come across the issue of wanting to adjust settings simultaneously for many layers, so I think this can take priority from a user-friendliness perspective.
- ditto for the UI audit, though perhaps it should wait until everything else in this roadmap, so maybe it can be jettisoned for now.
- serialization: I think 90%+ of use cases involved data writing, and that should be the focus, rather than things that are napari-specific. I think the use case of "I want to write my data to a portable format for further analysis, publication, archival, or to use in different software" is much more common than "I want to share this entire viewer including state with someone else". Indeed, with the former, the latter becomes quite a bit easier ("here's the data, can you open it with this script and go to slice 47?"), and everything else is convenience rather than necessity.
- headless: well, I'm still behind my mantra of "napari headless is just Python", though I understand that connecting to a remote napari is interesting. But, we have accessing remote data in the roadmap already. There's lots of options for that and I don't see that headless should get its own heading (heh) here.

Anyway, this is just my loosely-held perspective. We can discuss more during the meeting! Happy to be overruled if others feel differently.

Reply:
This additional section has been eliminated - UI and layer groups are now framed under making the viewer easier to use, serialization doesn't get mentioned, headless doesn't get mentioned, and CI has been moved under the eliminate all bugs.

I think this achieves the right balance between being user facing and motivated, but also letting developers know what is planned and prioritized

Reply:
This additional section has been eliminated - UI and layer groups are now framed under making the viewer easier to use, viewer serialization appears in "work for future roadmaps", headless only gets mentioned in the "viewing big datasets" , and CI has been moved under "eliminate all bugs".
Remote jupyter notebooks also now get mentioned in "work for future roadmaps"

I think this achieves the right balance between being user facing and motivated, but also letting developers know what is planned and prioritized and what will come later